components/brave_wallet/browser/internal/hd_key.cc

```diff
@@ -703,9 +687,56 @@ std::vector<uint8_t> HDKey::Sign(const std::vector<uint8_t>& msg, int* recid) {
   return sig;
 }
 
+// TODO(apaymyshev): return as base::expected?
+absl::optional<std::vector<uint8_t>> HDKey::SignDer(
+    base::span<const uint8_t, 32> msg) {
+  unsigned char extra_entropy[32] = {0};
+  secp256k1_ecdsa_signature ecdsa_sig;
+  if (!secp256k1_ecdsa_sign(secp256k1_ctx_, &ecdsa_sig, msg.data(),
+                            private_key_.data(),
+                            secp256k1_nonce_function_rfc6979, nullptr)) {
+    LOG(ERROR) << __func__ << ": secp256k1_ecdsa_sign failed";
+    return absl::nullopt;
+  }
+
+  auto sig_has_low_r = [](const secp256k1_context* ctx,
+                          const secp256k1_ecdsa_signature* sig) {
+    uint8_t compact_sig[kCompactSignatureSize] = {};
+    secp256k1_ecdsa_signature_serialize_compact(ctx, compact_sig, sig);
+
+    return compact_sig[0] < 0x80;
+  };
+
+  // Grind R https://github.com/bitcoin/bitcoin/pull/13666
+  uint32_t extra_entropy_counter = 0;
+  while (!sig_has_low_r(secp256k1_ctx_, &ecdsa_sig)) {
+    (*reinterpret_cast<uint32_t*>(extra_entropy)) =
+        base::ByteSwapToLE32(++extra_entropy_counter);
+
+    if (!secp256k1_ecdsa_sign(
+            secp256k1_ctx_, &ecdsa_sig, msg.data(), private_key_.data(),
+            secp256k1_nonce_function_rfc6979, extra_entropy)) {
```

Comment:
noting I've reviewed this [signing operation inner functionality](https://github.com/bitcoin-core/secp256k1/blob/master/src/secp256k1.c#L465-L570) to make sure that the extra_entropy is being fed in properly such that it's generating a new random K value for the signature, but doing so in a deterministic way still. It appears to be working as expected.